e2e/testcases/cli_test.go

```diff
@@ -1331,6 +1331,28 @@ func TestNomosMigrate(t *testing.T) {
 		},
 	}
 
+	nt.T.Cleanup(func() {
+		cmObj := &unstructured.Unstructured{
+			Object: map[string]interface{}{
+				"apiVersion": "configmanagement.gke.io/v1",
+				"kind":       "ConfigManagement",
+				"metadata": map[string]interface{}{
+					"name": "config-management",
+				},
+			},
+		}
+		// Delete the ConfigManagement CR, in case the test failed early.
+		if err := nt.KubeClient.Delete(cmObj); err != nil {
```

Comment:
Do we need to remove the finalizer before deleting the CR?

Reply:
In this case my intention was to just allow the finalizer to run, since the test manifests will be re-installed by the subsequent cleanup block

Reply:
If you prefer there's a few alternative approaches:
- Use `Apply` instead of `Create`, so it doesn't error if the object already exists (or other variations of not erroring if the object already exists)
- Change the cleanup ordering to delete the Deployment first, and then delete the ConfigManagement object and manually remove finalizers (otherwise the operator would re-apply the finalizers)

Reply:
I would prefer removing the Deployment first and then removing the finalizer and deleting the ConfigManagement object. My concern is that operator may fail to garbage collect resources in the cleanup stage.

Reply:
done